api/server.go

```diff
@@ -98,14 +99,25 @@ func (s *Server) walletAddressHandler(w http.ResponseWriter, req *http.Request,
 
 func (s *Server) walletAddressesHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
 	addresses := s.w.Addresses()
-	offset, limit := offsetLimit(req.URL.Query(), len(addresses))
-	writeJSON(w, WalletAddresses(addresses[offset:limit]))
+	begin, end := beginEnd(req.URL.Query(), len(addresses))
+	writeJSON(w, WalletAddresses(addresses[begin:end]))
 }
 
 func (s *Server) walletTransactionsHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
-	transactions := s.w.Transactions()
-	offset, limit := offsetLimit(req.URL.Query(), len(transactions))
-	writeJSON(w, transactions[offset:limit])
+	q := req.URL.Query()
+	since := q.Get("since")
+	sinceParsed, err := strconv.ParseInt(since, 10, 64)
+	if err != nil && since != "" {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	max := q.Get("max")
+	maxParsed, err := strconv.Atoi(max)
+	if err != nil && max != "" {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	writeJSON(w, s.w.Transactions(time.Unix(sinceParsed, 0), maxParsed))
```

Comment:
`maxParsed` should have a default value of `-1`, so that `Transactions` returns everything.